### PR TITLE
Fix 64

### LINKED ISSUE #64: Allow specification of multiple comparison groups
Show multiple volcano plots side-by-side for multiple. Highlight point on all charts when you hover on a point

### LINKED ISSUE #70: Remove the use of EnhancedVolcano
...from VolcanoPlot.R, assuming this isn't needed anymore.

### LINKED ISSUE #71: getStats `stat` argument defaults incorrect
Per roxygen should be "RR" and "RD" instead of "Risk Ratio" and "Risk Difference", which seems simpler for the user

### PR DESCRIPTION
**Pull request by anhtr**
Hello,

This pull request is an attempt to fix #64. The first step is to update getStats to take more than one Comparison groups in the setting, and to create a dataframe with a "comp_grp" column for future grouping/filtering in the plotting. The second step is to update volcanoPlot to make use of the new dataframe to plot side by side plots.

Sorry for the large amount of "changes". Most of them were because I used "Reformat code" in RStudio.

Thank you.

### CODE REVIEW

**Comment by anhtr** [comment]
(As of [5bd3186](https://github.com/SafetyGraphics/volcanoPlot/pull/69/commits/5bd3186f241e4457515f5c63ca1c2dac5bc67490))

So I implement this with crosstalk. I think it work decently well: two plots will be shown if two comparison groups are provided. If you click on a point in one graph, it will highlight the same point in the other. A thing I could not get it to do is to show the hover text on both graphs at the same time. However, you can make a compromise by clicking on one point in one graph and hover over the highlighted one in the other.

A current bug is that this is considered two plots, and currently would give an error if you run it in shiny app. I think either ui or server needs to be updated to handle the new output.

Test with this code:

```
settings<-list(
  stratification_col="AEBODSYS",
  group_col="ARM",
  reference_group="Placebo",
  comparison_group=c("Xanomeline High Dose","Xanomeline Low Dose"),
  id_col="USUBJID"
)
stats<-getStats(dfAE=safetyData::adam_adae, dfDemog = safetyData::adam_adsl, settings)
volcanoPlot(stats)
```

image

**Comment by anhtr** [comment]
I also removed the option to not use plotly, because it doesn't make sense anymore.

**Comment by anhtr** [comment]
Also for the next update: label the graphs with the comparison groups. I will work on it later and it should be simple enough that it won't break everything.

File: R/volcanoPlot.R [hunk 1]
```diff
@@ -14,18 +13,19 @@
...
-#'   comparison_group="Xanomeline High Dose",
+#'   comparison_group=c("Xanomeline High Dose","Xanomeline Low Dose"),
 #'   id_col="USUBJID"
 #' )
 #' stats<-getStats(dfAE=safetyData::adam_adae, dfDemog = safetyData::adam_adsl, settings)
 #' volcanoPlot(stats)
 #'
 #' @import ggplot2
 #' @importFrom plotly ggplotly
+#' @import crosstalk
```
Comment (bzkrouse) [inline R/volcanoPlot.R:24]:
```suggestion
#' @importFrom crosstalk SharedData bscols
```

Comment (bzkrouse) [reply R/volcanoPlot.R:24]: Only importing what we need will help us to avoid the following warning:

```Warning message:
replacing previous import ‘crosstalk::getDefaultReactiveDomain’ by ‘shiny::getDefaultReactiveDomain’ when loading ‘volcanoPlot’ ```

File: R/getStats.R [hunk 2]
```diff
@@ -27,70 +27,87 @@ getStats <- function(dfAE, dfDemog, settings, stat="Risk Ratio") {
     print(settings)
```
Comment (bzkrouse) [inline R/getStats.R:27]:
```suggestion
```

File: R/volcano_server.R [hunk 3]
```diff
@@ -38,15 +38,54 @@ volcano_server <- function(input, output, session, params) {
...
+            stats()[plotly_d$pointNumber+1,]
+        }
+    })
+
+    ae_sub <- reactive({
+        all<-params()$data$aes
+        # subset to selected strata
+        sub1<-all[all[params()$settings$aes$bodsys_col]==stats_sub()$strata,]
+
+        # subset to selected groups
```
Comment (bzkrouse) [inline R/volcano_server.R:71]: Is it okay to delete this commented out code?

**Review by bzkrouse — CHANGES_REQUESTED** [review]
@anhtr Great progress! I added some comments in the PR.
As you mentioned, there is a need to add the option to toggle the comparison group to the UI. We can discuss this further in our meeting, but here are some initial thoughts:

It looks like volcanoApp() receives information about the treatment groups from the settings object (currently the expectation is there will be a "group1" (reference group) and "group2" (comparison group) - see lines 24 & 25 of volcano_server.R.  

If we keep the settings object as-is, the user could set settings$group2 to be a vector of multiple choices. These would then populate a dropdown in the UI. The user selection would then be passed to line 25 in the volcano_server. 
However I might suggest we rename group1 and group2 to be something more informative.

-----------------------------------------------------------------------------------------------------
To recap group discussion:
- No UI dropdown needed
- settings to specify the reference group but just assume the rest of the trt groups will be comparison groups
- Display all comparisons

**Comment by anhtr** [comment]
I have updated the package:

- If setting not specified, the default reference group is 'Placebo' and everything else will be comparison group. As the result:
  + getStats() was updated to put group names directly in the output table.
  + volcanoPlot() pulls group names directly out of the stats table.
  + Anything that uses groups() reactive is now updated to stop using it, as it does not make sense anymore.
  + We no longer filter for groups in the click table.
- Arbitrary number of plots should display correctly in the UI.
- EnhancedVolcano plot was removed. https://github.com/SafetyGraphics/volcanoPlot/issues/70
- Minimal comments were added to clarify functions.

**Comment by jwildfire** [comment]
Closing this one since changes are captured in #74 also
